src/test/suite/fixtures/recorded/languages/rust/changeState2.yml

```diff
@@ -0,0 +1,36 @@
+languageId: rust
+command:
+  spokenForm: change state
+  version: 2
+  targets:
+    - type: primitive
+      modifiers:
+        - type: containingScope
+          scopeType: {type: statement}
+  usePrePhraseSnapshot: true
+  action: {name: clearAndSetSelection}
+initialState:
+  documentContents: |
+    let expensive_closure = |num| {
+        println!("calculating slowly...");
+        thread::sleep(Duration::from_secs(2));
+        num
+    };
+  selections:
+    - anchor: {line: 1, character: 15}
+      active: {line: 1, character: 15}
+  marks: {}
+finalState:
+  documentContents: |
+    let expensive_closure = |num| {
+        ;
```

Comment:
That may well be, but the parser views it in a fairly strange way, so I had assumed it was considered non-idiomatic:

```
source_file [0, 0] - [3, 0]
  if_expression [0, 0] - [2, 1]
    condition: boolean_literal [0, 3] - [0, 7]
    consequence: block [0, 8] - [2, 1]
      macro_invocation [1, 4] - [1, 21]
        macro: identifier [1, 4] - [1, 11]
        token_tree [1, 12] - [1, 21]
          string_literal [1, 13] - [1, 20]
  empty_statement [2, 1] - [2, 2]
```

Reply:
I’m not sure I get what you want here. If you want semicolon to be removed when we change statement then I’m not sure how to achieve that. Do I need to wrap ancestorChain matcher into trailing matcher with semicolon or something?

Reply:
ah wait a minute; that semicolon is actually extra isn't it?  because `println!` is a macro, and macros don't actually require a semicolon.  This is fine